<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
<title>NITRO/CODA C++ Modules: except::Error Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css">
<link href="doxygen.css" rel="stylesheet" type="text/css">
</head><body>
<!-- Generated by Doxygen 1.5.8 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="main.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="classes.html"><span>Class&nbsp;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath"><a class="el" href="namespaceexcept.html">except</a>::<a class="el" href="classexcept_1_1Error.html">Error</a>
  </div>
</div>
<div class="contents">
<h1>except::Error Class Reference</h1><!-- doxytag: class="except::Error" --><!-- doxytag: inherits="except::Throwable" -->Represents a serious unexpected occurrence during the program.  
<a href="classexcept_1_1Error.html#_details">More...</a>
<p>
<code>#include &lt;<a class="el" href="Error_8h-source.html">Error.h</a>&gt;</code>
<p>
<div class="dynheader">
Inheritance diagram for except::Error:</div>
<div class="dynsection">

<p><center><img src="http://nitro-nitf.sourceforge.net/apidocs/c/2.5cpp/classexcept_1_1Error.png" usemap="#except::Error_map" border="0" alt=""></center>
<map name="except::Error_map">
<area href="classexcept_1_1Throwable.html" alt="except::Throwable" shape="rect" coords="102,0,297,24">
<area href="classexcept_1_1InvalidDerivedTypeError.html" alt="except::InvalidDerivedTypeError" shape="rect" coords="0,112,195,136">
<area href="classsys_1_1SystemError.html" alt="sys::SystemError" shape="rect" coords="205,112,400,136">
</map>
</div>

<p>
<a href="http://nitro-nitf.sourceforge.net/apidocs/c/2.5cpp/classexcept_1_1Error-members.html">List of all members.</a><table border="0" cellpadding="0" cellspacing="0">
<tr><td></td></tr>
<tr><td colspan="2"><br><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Error.html#6a43b267b5e2293bec22f72377671161">Error</a> ()</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Error.html#7a105256237ccf3074b75364de5af8cf">Error</a> (const <a class="el" href="classexcept_1_1Context.html">Context</a> &amp;c)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Error.html#cd7f72d2e197bfde76201965d6c6c375">Error</a> (const std::string &amp;message)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Error.html#7f6b0f74ad07bd2ddb9402758f9f7fb6">Error</a> (const <a class="el" href="classexcept_1_1Throwable.html">Throwable</a> &amp;t, const <a class="el" href="classexcept_1_1Context.html">Context</a> &amp;c)</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Error.html#8b7d91496b18583b83862be94f2b631e">~Error</a> ()</td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Destructor.  <a href="classexcept_1_1Error.html#8b7d91496b18583b83862be94f2b631e"></a><br></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">std::string&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Throwable.html#0971ae70abf07e199783eab24ead41d2">getMessage</a> () const </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">const <a class="el" href="classexcept_1_1Trace.html">Trace</a> &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Throwable.html#47d1c81e42b7130dd9819b658c2e1f77">getTrace</a> () const </td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="el" href="classexcept_1_1Trace.html">Trace</a> &amp;&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Throwable.html#bcf1803c21f8d2100b6caced73263ab2">getTrace</a> ()</td></tr>

<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual const char *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Throwable.html#aa4376fa39ef90e058ef26527ee06e9b">getType</a> () const </td></tr>

<tr><td colspan="2"><br><h2>Protected Attributes</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top"><a class="el" href="classexcept_1_1Trace.html">Trace</a>&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Throwable.html#3814d2d339c0cee2afc2ae29765f173e">mTrace</a></td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">The name of exception trace.  <a href="classexcept_1_1Error.html#3814d2d339c0cee2afc2ae29765f173e"></a><br></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">std::string&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classexcept_1_1Throwable.html#4eb7c05c2e676ea0546138a562e915ab">mMessage</a></td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">The name of the message the exception was thrown.  <a href="classexcept_1_1Error.html#4eb7c05c2e676ea0546138a562e915ab"></a><br></td></tr>
</table>
<hr><a name="_details"></a><h2>Detailed Description</h2>
Represents a serious unexpected occurrence during the program. 
<p>
The error class is a representation of a throwable object, occurring under serious conditions. It may be undesirable to handle an error in the manner that you handle an exception. For this reason, the distinction is made 
<p>Definition at line <a class="el" href="Error_8h-source.html#l00050">50</a> of file <a class="el" href="Error_8h-source.html">Error.h</a>.</p>
<hr><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" name="6a43b267b5e2293bec22f72377671161"></a><!-- doxytag: member="except::Error::Error" ref="6a43b267b5e2293bec22f72377671161" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">except::Error::Error           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Default constructor 
<p>Definition at line <a class="el" href="Error_8h-source.html#l00056">56</a> of file <a class="el" href="Error_8h-source.html">Error.h</a>.</p>

</div>
</div><p>
<a class="anchor" name="7a105256237ccf3074b75364de5af8cf"></a><!-- doxytag: member="except::Error::Error" ref="7a105256237ccf3074b75364de5af8cf" args="(const Context &amp;c)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">except::Error::Error           </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classexcept_1_1Context.html">Context</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>c</em>          </td>
          <td>&nbsp;)&nbsp;</td>
          <td><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Constructor. Takes a <a class="el" href="classexcept_1_1Context.html" title="The information surrounding an exception or error.">Context</a> <dl compact><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>c</em>&nbsp;</td><td>The <a class="el" href="classexcept_1_1Context.html" title="The information surrounding an exception or error.">Context</a> </td></tr>
  </table>
</dl>

<p>Definition at line <a class="el" href="Error_8h-source.html#l00063">63</a> of file <a class="el" href="Error_8h-source.html">Error.h</a>.</p>

</div>
</div><p>
<a class="anchor" name="cd7f72d2e197bfde76201965d6c6c375"></a><!-- doxytag: member="except::Error::Error" ref="cd7f72d2e197bfde76201965d6c6c375" args="(const std::string &amp;message)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">except::Error::Error           </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>message</em>          </td>
          <td>&nbsp;)&nbsp;</td>
          <td><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Constructor. Takes a message <dl compact><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>message</em>&nbsp;</td><td>The message </td></tr>
  </table>
</dl>

<p>Definition at line <a class="el" href="Error_8h-source.html#l00070">70</a> of file <a class="el" href="Error_8h-source.html">Error.h</a>.</p>

</div>
</div><p>
<a class="anchor" name="7f6b0f74ad07bd2ddb9402758f9f7fb6"></a><!-- doxytag: member="except::Error::Error" ref="7f6b0f74ad07bd2ddb9402758f9f7fb6" args="(const Throwable &amp;t, const Context &amp;c)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">except::Error::Error           </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classexcept_1_1Throwable.html">Throwable</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>t</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const <a class="el" href="classexcept_1_1Context.html">Context</a> &amp;&nbsp;</td>
          <td class="paramname"> <em>c</em></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td><code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Constructor. Takes an <a class="el" href="classexcept_1_1Throwable.html" title="The interface for exceptions and errors.">Throwable</a> and a <a class="el" href="classexcept_1_1Context.html" title="The information surrounding an exception or error.">Context</a> <dl compact><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>t</em>&nbsp;</td><td>The <a class="el" href="classexcept_1_1Throwable.html" title="The interface for exceptions and errors.">Throwable</a> </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>c</em>&nbsp;</td><td>The <a class="el" href="classexcept_1_1Context.html" title="The information surrounding an exception or error.">Context</a> </td></tr>
  </table>
</dl>

<p>Definition at line <a class="el" href="Error_8h-source.html#l00078">78</a> of file <a class="el" href="Error_8h-source.html">Error.h</a>.</p>

</div>
</div><p>
<a class="anchor" name="8b7d91496b18583b83862be94f2b631e"></a><!-- doxytag: member="except::Error::~Error" ref="8b7d91496b18583b83862be94f2b631e" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual except::Error::~Error           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td><code> [inline, virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Destructor. 
<p>

<p>Definition at line <a class="el" href="Error_8h-source.html#l00087">87</a> of file <a class="el" href="Error_8h-source.html">Error.h</a>.</p>

</div>
</div><p>
<hr><h2>Member Function Documentation</h2>
<a class="anchor" name="0971ae70abf07e199783eab24ead41d2"></a><!-- doxytag: member="except::Error::getMessage" ref="0971ae70abf07e199783eab24ead41d2" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string except::Throwable::getMessage           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [inline, inherited]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Get the message <dl class="return" compact><dt><b>Returns:</b></dt><dd>The message </dd></dl>

<p>Definition at line <a class="el" href="Throwable_8h-source.html#l00091">91</a> of file <a class="el" href="Throwable_8h-source.html">Throwable.h</a>.</p>

<p>References <a class="el" href="Throwable_8h-source.html#l00127">except::Throwable::mMessage</a>.</p>

</div>
</div><p>
<a class="anchor" name="bcf1803c21f8d2100b6caced73263ab2"></a><!-- doxytag: member="except::Error::getTrace" ref="bcf1803c21f8d2100b6caced73263ab2" args="()" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classexcept_1_1Trace.html">Trace</a>&amp; except::Throwable::getTrace           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td><code> [inline, inherited]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Get the trace <dl class="return" compact><dt><b>Returns:</b></dt><dd>The trace (non-const) </dd></dl>

<p>Definition at line <a class="el" href="Throwable_8h-source.html#l00109">109</a> of file <a class="el" href="Throwable_8h-source.html">Throwable.h</a>.</p>

<p>References <a class="el" href="Throwable_8h-source.html#l00125">except::Throwable::mTrace</a>.</p>

</div>
</div><p>
<a class="anchor" name="47d1c81e42b7130dd9819b658c2e1f77"></a><!-- doxytag: member="except::Error::getTrace" ref="47d1c81e42b7130dd9819b658c2e1f77" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const <a class="el" href="classexcept_1_1Trace.html">Trace</a>&amp; except::Throwable::getTrace           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [inline, inherited]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Get the trace <dl class="return" compact><dt><b>Returns:</b></dt><dd>The trace (const) </dd></dl>

<p>Definition at line <a class="el" href="Throwable_8h-source.html#l00100">100</a> of file <a class="el" href="Throwable_8h-source.html">Throwable.h</a>.</p>

<p>References <a class="el" href="Throwable_8h-source.html#l00125">except::Throwable::mTrace</a>.</p>

</div>
</div><p>
<a class="anchor" name="aa4376fa39ef90e058ef26527ee06e9b"></a><!-- doxytag: member="except::Error::getType" ref="aa4376fa39ef90e058ef26527ee06e9b" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual const char* except::Throwable::getType           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [inline, virtual, inherited]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Get the type id <dl class="return" compact><dt><b>Returns:</b></dt><dd>The type </dd></dl>

<p>Definition at line <a class="el" href="Throwable_8h-source.html#l00118">118</a> of file <a class="el" href="Throwable_8h-source.html">Throwable.h</a>.</p>

</div>
</div><p>
<hr><h2>Member Data Documentation</h2>
<a class="anchor" name="4eb7c05c2e676ea0546138a562e915ab"></a><!-- doxytag: member="except::Error::mMessage" ref="4eb7c05c2e676ea0546138a562e915ab" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string <a class="el" href="classexcept_1_1Throwable.html#4eb7c05c2e676ea0546138a562e915ab">except::Throwable::mMessage</a><code> [protected, inherited]</code>          </td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
The name of the message the exception was thrown. 
<p>

<p>Definition at line <a class="el" href="Throwable_8h-source.html#l00127">127</a> of file <a class="el" href="Throwable_8h-source.html">Throwable.h</a>.</p>

<p>Referenced by <a class="el" href="Throwable_8h-source.html#l00091">except::Throwable::getMessage()</a>, <a class="el" href="Exception_8h-source.html#l00209">except::IndexOutOfRangeException::IndexOutOfRangeException()</a>, <a class="el" href="NITFException_8hpp-source.html#l00046">nitf::NITFException::NITFException()</a>, <a class="el" href="SystemException_8h-source.html#l00121">sys::SystemError::SystemError()</a>, and <a class="el" href="SystemException_8h-source.html#l00052">sys::SystemException::SystemException()</a>.</p>

</div>
</div><p>
<a class="anchor" name="3814d2d339c0cee2afc2ae29765f173e"></a><!-- doxytag: member="except::Error::mTrace" ref="3814d2d339c0cee2afc2ae29765f173e" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classexcept_1_1Trace.html">Trace</a> <a class="el" href="classexcept_1_1Throwable.html#3814d2d339c0cee2afc2ae29765f173e">except::Throwable::mTrace</a><code> [protected, inherited]</code>          </td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
The name of exception trace. 
<p>

<p>Definition at line <a class="el" href="Throwable_8h-source.html#l00125">125</a> of file <a class="el" href="Throwable_8h-source.html">Throwable.h</a>.</p>

<p>Referenced by <a class="el" href="Throwable_8h-source.html#l00100">except::Throwable::getTrace()</a>, and <a class="el" href="NITFException_8hpp-source.html#l00046">nitf::NITFException::NITFException()</a>.</p>

</div>
</div><p>
<hr>The documentation for this class was generated from the following file:<ul>
<li><a class="el" href="Error_8h-source.html">Error.h</a></ul>
</div>
<hr size="1"><address style="text-align: right;"><small>Generated on Wed Dec 9 19:35:22 2009 for NITRO/CODA C++ Modules by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 1.5.8 </small></address>
</body>
</html>
